selfdrive/car/toyota/carstate.py

```diff
@@ -94,6 +94,9 @@ def update(self, cp, cp_cam):
       ret.cruiseState.available = cp.vl["PCM_CRUISE_2"]["MAIN_ON"] != 0
       ret.cruiseState.speed = cp.vl["PCM_CRUISE_2"]["SET_SPEED"] * CV.KPH_TO_MS
 
+    # TODO: where is the metric/imperial bit?
```

Comment:
`UI_SETTING->UNITS`?

Reply:
That's potentially linked to the radio, so it might not be present depending on the configuration of the vehicle

Reply:
I checked all our test routes in CI (one route per platform) and they all have the UI_SETTING message. Are you sure?

Reply:
Ah, the message may exist, but the UNITS signal is 0 on some cars.

Reply:
So I found `RSA2->SPDUNT`, and a new undefined signal on address 0x435, msb: 55, size: 3 which not only has the units, but the preference of gas mileage (however this didn't exist on the TSS-P RAV4, just my TSS2.5 Camry Hybrid)

Reply:
Wait a minute, I checked the 2018 Camry route, and both `UI_SETTING->UNITS` and `RSA2->SPDUNT` were 0, *however* the new signal I found at 0x435 *does* exist, and it's 3, indicating the user had it set to imperial. So it looks like we can't use RSA2 or UI_SETTING across the board sadly. We might have to detect the presence of each message

Reply:
Can't believe I missed this! Found a common units signal